player/player.go

```diff
@@ -0,0 +1,227 @@
+package player
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"sort"
+	"strconv"
+
+	"github.com/lbryio/lbry.go/extras/errors"
+	ljsonrpc "github.com/lbryio/lbry.go/extras/jsonrpc"
+	"github.com/lbryio/lbry.go/stream"
+	"github.com/lbryio/lbryweb.go/config"
+	"github.com/lbryio/lbryweb.go/monitor"
+	log "github.com/sirupsen/logrus"
+)
+
+const reflectorURL = "http://blobs.lbry.io/"
+
+type reflectedStream struct {
+	URI         string          `json:"-"`
+	SDHash      []byte          `json:"stream_hash"`
+	Blobs       []reflectedBlob `json:"blobs"`
+	Key         string          `json:"key"`
+	DecodedKey  []byte          `json:"-"`
+	Size        int             `json:"-"`
+	ContentType string          `json:"-"`
+}
+
+type reflectedBlob struct {
+	Hash          string `json:"blob_hash"`
+	Number        int    `json:"blob_num"`
+	IV            string `json:"iv"`
+	DecodedIV     []byte `json:"-"`
+	Length        int    `json:"length"`
+	encryptedBody []byte
+}
+
+// PlayURI downloads and streams LBRY video content located at uri delimited by rangeHeader
+// (use rangeHeader := request.Header.Get("Range"))
+func PlayURI(uri string, rangeHeader string, writer http.ResponseWriter) (err error) {
+	start, end := parseRange(rangeHeader)
+	stream := newReflectedStream(uri)
+	err = stream.resolve()
+	if err != nil {
+		return err
+	}
+	err = stream.fetchData()
+	if err != nil {
+		return err
+	}
+	blobStart, blobEnd := stream.prepareWriter(start, end, writer)
+	err = stream.streamBlobs(blobStart, blobEnd, writer)
+	return err
+}
+
+func parseRange(header string) (int64, int64) {
+	r := regexp.MustCompile(`bytes=(\d+)-(\d*)`)
+	m := r.FindStringSubmatch(header)
+	if len(m) == 0 {
+		return 0, 0
+	}
+	start, err1 := strconv.ParseInt(m[1], 10, 64)
+	end, err2 := strconv.ParseInt(m[2], 10, 64)
+	if err1 != nil || err2 != nil {
+		return 0, 0
+	}
+	if start < 0 {
+		start = 0
+	}
+	if end < 0 {
+		end = 0
+	}
+	if start > end {
+		start = 0
+		end = 0
+	}
+	return start, end
+}
+
+func newReflectedStream(uri string) *reflectedStream {
+	return &reflectedStream{URI: uri}
+}
+
+func (s *reflectedStream) URL() string {
+	return reflectorURL + string(s.SDHash)
+}
+
+// resolve gets SD hash from the daemon for a given URI to enable further retrieval of stream content from
+// reflector servers
+func (s *reflectedStream) resolve() error {
+	client := ljsonrpc.NewClient(config.Settings.GetString("Lbrynet"))
```

Comment:
i recommend passing this value into newReflectedStream() instead of using a global config